### PR TITLE
feat: allow overriding YamlDotNet settings

### PR DESCRIPTION
**Pull request by IvanJosipovic**
This PR allows overriding the default YamlDotNet settings by adding a new KubernetesYamlConfiguration class.

I'm not able to test this as its static.

### CODE REVIEW

File: src/KubernetesClient.Models/KubernetesYamlConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace k8s
+{
+    /// <summary>
+    /// This class allows overriding the <see cref="YamlDotNet"/> settings.
+    /// </summary>
+    public static class KubernetesYamlConfiguration
+    {
+        internal static Action<DeserializerBuilder> DeseralizerAction { get; set; }
+        internal static Action<SerializerBuilder> SeralizerAction { get; set; }
```
Comment (tg123) [inline src/KubernetesClient.Models/KubernetesYamlConfiguration.cs:11]: event?

File: src/KubernetesClient.Models/KubernetesYamlConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        /// </summary>
+        /// <param name="configure">An <see cref="Action"/> to configure the <see cref="SerializerBuilder"/>.</param>
+        public static void AddSerializerOptions(Action<SerializerBuilder> configure)
+        {
+            if (configure is null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            SeralizerAction = configure;
```
Comment (tg123) [inline src/KubernetesClient.Models/KubernetesYamlConfiguration.cs:24]:
```suggestion
            SeralizerAction += configure;
```

**Review by tg123 — COMMENTED** [review]
maybe you want event?

**Comment by IvanJosipovic** [comment]
I converted the logic to use an event. It can be tested like this:

```c#
using k8s;
using System.Text.Json.Nodes;
using YamlDotNet.Serialization;
using YamlDotNet.System.Text.Json;

KubernetesYamlConfiguration.SeralizerEvent += KubernetesYamlConfiguration_SeralizerEvent;

void KubernetesYamlConfiguration_SeralizerEvent(object? sender, SerializerBuilder e)
{
    e.WithTypeConverter(new SystemTextJsonYamlTypeConverter(true))
     .WithTypeInspector(x => new SystemTextJsonTypeInspector(x));
}

KubernetesYamlConfiguration.DeseralizerEvent += KubernetesYamlConfiguration_DeseralizerEvent;

void KubernetesYamlConfiguration_DeseralizerEvent(object? sender, DeserializerBuilder e)
{
    e.WithTypeConverter(new SystemTextJsonYamlTypeConverter())
    .WithTypeInspector(x => new SystemTextJsonTypeInspector(x));
}

var obj = KubernetesYaml.LoadFromString<JsonObject>(File.ReadAllText("helm-release.yaml"));

var yaml = KubernetesYaml.Serialize(obj);

Console.ReadLine();
```

**Comment by tg123** [comment]
close as of #1498

**Comment by IvanJosipovic** [comment]
@tg123, can we please re-open this PR?
This PR is not related to AOT. It allows customizing the YAML serializer settings, this library has a similar capability for JSON.

**Pull request closed without merging**